integration/users/usersinfo_test.go

```diff
@@ -0,0 +1,434 @@
+						{"user", "i"},
+						{"db", "TestUsersinfo_few"},
+						{"roles", bson.A{}},
+					},
+					bson.D{
+						{"_id", "TestUsersinfo.one"},
+						{"user", "one"},
+						{"db", "TestUsersinfo"},
+						{"roles", bson.A{}},
+					},
+				}},
+				{"ok", float64(1)},
+			},
+		},
+		"FromMixed": {
+			dbSuffix: "_few",
+			payload: bson.D{{
+				"usersInfo", bson.A{
+					bson.D{
+						{"user", "one"},
+						{"db", "TestUsersinfo"},
+					},
+					"i",
+				},
+			}},
+			expected: bson.D{
+				{"users", bson.A{
+					bson.D{
+						{"_id", "TestUsersinfo_few.i"},
+						{"user", "i"},
+						{"db", "TestUsersinfo_few"},
+						{"roles", bson.A{}},
+					},
+					bson.D{
+						{"_id", "TestUsersinfo.one"},
+						{"user", "one"},
+						{"db", "TestUsersinfo"},
+						{"roles", bson.A{}},
+					},
+				}},
+				{"ok", float64(1)},
+			},
+		},
+		"Many": {
+			dbSuffix: "_example",
+			payload: bson.D{
+				{"usersInfo", 1},
+			},
+			expected: bson.D{
+				{"users", bson.A{
+					bson.D{
+						{"_id", "TestUsersinfo_example.a"},
+						{"user", "a"},
+						{"db", "TestUsersinfo_example"},
+						{"roles", bson.A{}},
+					},
+					bson.D{
+						{"_id", "TestUsersinfo_example.b"},
+						{"user", "b"},
+						{"db", "TestUsersinfo_example"},
+						{"roles", bson.A{}},
+					},
+					bson.D{
+						{"_id", "TestUsersinfo_example.c"},
+						{"user", "c"},
+						{"db", "TestUsersinfo_example"},
+						{"roles", bson.A{}},
+					},
+				}},
+				{"ok", float64(1)},
+			},
+		},
+		"Multiple": {
+			dbSuffix: "_example",
+			payload: bson.D{
+				{"usersInfo", bson.A{
+					bson.D{{"user", "b"}, {"db", "TestUsersinfo_example"}},
+					bson.D{{"user", "c"}, {"db", "TestUsersinfo_example"}},
+				}},
+			},
+			expected: bson.D{
+				{"users", bson.A{
+					bson.D{
+						{"_id", "TestUsersinfo_example.b"},
+						{"user", "b"},
+						{"db", "TestUsersinfo_example"},
+						{"roles", bson.A{}},
+					},
+					bson.D{
+						{"_id", "TestUsersinfo_example.c"},
+						{"user", "c"},
+						{"db", "TestUsersinfo_example"},
+						{"roles", bson.A{}},
+					},
+				}},
+				{"ok", float64(1)},
+			},
+		},
+		"MissingDBFieldName": {
+			dbSuffix: "_example",
+			payload: bson.D{
+				{"usersInfo", bson.A{
+					bson.D{{"user", "missing_db"}},
+				}},
+			},
+			err: &mongo.CommandError{
+				Code:    2,
+				Message: "UserName must contain a field named: db",
+				Name:    "BadValue",
+			},
+		},
+		"ForAllDBs": {
+			dbSuffix: "_example",
+			payload: bson.D{{
+				"usersInfo", bson.D{
+					{"forAllDBs", true},
+				},
+			}},
+			hasUser: map[string]struct{}{
+				"TestUsersinfo.one":                {},
+				"TestUsersinfo.two":                {},
+				"TestUsersinfo_example.a":          {},
+				"TestUsersinfo_example.b":          {},
+				"TestUsersinfo_example.c":          {},
+				"TestUsersinfo_few.i":              {},
+				"TestUsersinfo_few.j":              {},
+				"TestUsersinfo_another.singleuser": {},
+			},
+		},
+	}
```

Comment:
Maybe one more case (or a few cases) with `payload` being set as something like

```
"usersInfo", bson.D{{"foo", "bar"}}
```

and some variations of it - the cases when `usersInfo` is set and is a document (or an array of documents) but `user` field is not set in this document, or `db` is not set or some combinations of it.

Reply:
Added a few extra test cases.

A question: should we fail if there is an unknown field in the payload? I've noticed MongoDB fails if this happens on the usersInfo command. There isn't an API on *types.Document to check for this error, though 🤔

Reply:
Well, if MongoDB returns an error, it sounds reasonable for us to return an error too :)